en/concept/events_interface.md

```diff
@@ -0,0 +1,140 @@
+To use the API, add this include:
+```cpp
+#include <px4_platform_common/events.h>
+```
+And then define and send the event from the desired code location:
+```cpp
+events::send(events::ID("mymodule_test"), events::Log::Info, "Test Message");
+```
+
+#### Backward compatibility
+
+For older GCS versions without events interface support, PX4 currently sends out all events also as `mavlink_log_*` `STATUSTEXT` message.
+In addition, the message must be tagged with an appended tab (`\t`) so that newer GCS's can ignore that and only show the event.
+
+So whenever adding an event, be sure to also add a `mavlink_log_` call. For example:
+```cpp
+mavlink_log_info(mavlink_log_pub, "Test Message\t");
+events::send(events::ID("mymodule_test"), events::Log::Info, "Test Message");
+```
+All such `mavlink_log_` calls will be removed after the next release.
+
+### Detailed
+
+The above is a minimal example, this is a more extensive one:
+```cpp
+uint8_t arg1 = 0;
+float arg2 = -1.f;
+/* EVENT
+ * @description
+ * This is the detailed event description.
+ *
+ * - value of arg1: {1}
+ * - value of arg2: {2:.1}
+ *
+ * <profile name="dev">
+ * (This paragraph is only meant to be shown to developers).
+ * This behavior can be configured with the parameter <param>COM_EXAMPLE</param>.
+ * </profile>
+ *
+ * Link to documentation: <a>https://docs.px4.io</a>
+ */
+events::send<uint8_t, float>(events::ID("event_name"),
+	{events::Log::Error, events::LogInternal::Info}, "Event Message", arg1, arg2);
+```
+
+Explanations and requirements:
+- `/* EVENT`: this is an indicator for the parser that there is additional metadata following as comment.
+- **event_name**: the event name.
+  - must be unique within the whole source code of PX4.
+    As a general convention, prefix it with the module name, or the source file for larger modules.
+  - must be a valid variable name, i.e. must not contain spaces, colons, etc.
+  - from that name, a 24 bit event ID is derived using a hash function.
+    This means as long as the event name stays the same, so will the ID.
+- **Log Level**:
+  - valid log levels are the same as used in the MAVLink [MAV_SEVERITY](https://mavlink.io/en/messages/common.html#MAV_SEVERITY) enum.
+    In order of descending importance these are:
+    ```
+	Emergency,
+	Alert,
+	Critical,
+	Error,
+	Warning,
+	Notice,
+	Info,
+	Debug,
+	Disabled,
+	```
+  - Above we specify a separate external and internal log level, which are the levels displayed to GCS users and in the log file, respectively: `{events::Log::Error, events::LogInternal::Info}`.
+    For the majority of cases you can pass a single log level, and this will be used for both exernal and internal cases.
+	There are cases it makes sense to have two different log levels.
+	For example an RTL failsafe action: the user should see it as Warning/Error, whereas in the log, it is an expected system response, so it can be set to `Info`.
+- **Event Message**:
+  - Single-line, short description of the event.
```

Comment:
is this `@description`. Might be worth mentioning that.

Reply:
No, `Event Description` is the description.